File: tests/statistical_tests/spot_checks/mann-whitney.test.ts
import { stats } from "../../../src/dataframe/mod.ts";

Deno.test("Mann-Whitney U Test Spot Check", () => {
  console.log("\n📊 Mann-Whitney U Test");
  console.log("-".repeat(80));

  const group1 = [14, 15, 16, 17, 18];
  const group2 = [20, 21, 22, 23, 24];

  const result = stats.test.nonparametric.mannWhitney({
    x: group1,
    y: group2,
    alternative: "two-sided",
  });

  console.log("Group 1:", group1);
  console.log("Group 2:", group2);
  console.log("U-statistic:", result.test_statistic.value);
  console.log("p-value:", result.p_value);
});

console.log("\n" + "=".repeat(80));
console.log("MANN-WHITNEY U TEST SPOT CHECK");
console.log("=".repeat(80));

File: tests/statistical_tests/spot_checks/mann-whitney.test.R
#!/usr/bin/env Rscript

cat("\n📊 Mann-Whitney U Test\n")
cat(paste(rep("-", 80), collapse = ""), "\n")

group1 <- c(14, 15, 16, 17, 18)
group2 <- c(20, 21, 22, 23, 24)

# Mann-Whitney U test (Wilcoxon rank-sum test)
result <- wilcox.test(group1, group2, alternative = "two.sided", conf.level = 0.95)

cat("Group 1:", group1, "\n")
cat("Group 2:", group2, "\n")
cat("W-statistic:", result$statistic, "\n")
cat("p-value:", result$p.value, "\n")

cat("\n", paste(rep("=", 80), collapse = ""), "\n")
cat("MANN-WHITNEY U TEST SPOT CHECK\n")
cat(paste(rep("=", 80), collapse = ""), "\n")
